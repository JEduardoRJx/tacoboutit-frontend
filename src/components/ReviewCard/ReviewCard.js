import React, { Component } from 'react'
import { Text, View } from 'react-native'

export default class ReviewCard extends Component {
  renderReviews = () => {
    const reviews = this.props.taco.reviews;
    return reviews.map((review, i) => (
      <View key={`${review.review}${review.id}`}>
        <Text>Rating: {review.rating}/10</Text>
        <Text>{review.review}</Text>
      </View>
    ));
  }

  render() {
    const { taco } = this.props;
    console.log(taco);
    return (
      <View>
        <Text>Type: {taco.type}</Text>
        { taco.average_rating === null ? <Text>No rating yet</Text> : <Text>Average Rating: {taco.average_rating}/10</Text>}
        {!this.props.taco.reviews.length ? <Text>Be the first to add a review</Text> : this.renderReviews()}
      </View>
    )
  }
}
