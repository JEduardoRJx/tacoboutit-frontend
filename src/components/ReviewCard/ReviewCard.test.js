import React from 'react';
import ReviewCard from './ReviewCard';
import { shallow } from 'enzyme';



describe('ReviewCard', () => {
  const mockTaco = {
    "id": 9,
    "type": "Testing testing",
    "restaurant_id": 2,
    "average_rating": 5.0,
    "reviews": [
      {
        "id": 1,
        "rating": 7.0,
        "review": "its ok",
        "taco": 9
      },
      {
        "id": 2,
        "rating": 4.0,
        "review": "lame",
        "taco": 9
      },
      {
        "id": 3,
        "rating": 4.0,
        "review": null,
        "taco": 9
      },
      {
        "id": 4,
        "rating": 1.0,
        "review": "This taco put my son in the hospital",
        "taco": 9
      },
      {
        "id": 5,
        "rating": 10.0,
        "review": "The best",
        "taco": 9
      },
      {
        "id": 6,
        "rating": 4.0,
        "review": "meh",
        "taco": 9
      }
    ]
  };
  const mockTacoNoRatingsReviews = {
    "id": 9,
    "type": "Testing testing",
    "restaurant_id": 2,
    "average_rating": null,
    "reviews": [],
  };
  const wrapper = shallow(<ReviewCard taco={mockTaco} />);
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renderReviews should return an array of review elements', () => {
    const expected = [
      '<View><Text>Rating: 7/10</Text><Text>its ok</Text></View>',
      '<View><Text>Rating: 4/10</Text><Text>lame</Text></View>',
      '<View><Text>Rating: 4/10</Text><Text></Text></View>',
      '<View><Text>Rating: 1/10</Text><Text>This taco put my son in the hospital</Text></View>',
      '<View><Text>Rating: 10/10</Text><Text>The best</Text></View>',
      '<View><Text>Rating: 4/10</Text><Text>meh</Text></View>'
    ];
    const elements = wrapper.instance().renderReviews();
    expect(elements.length).toEqual(6);
    // const result = wrapper
    // .instance()
    // .renderReviews()
    // .map(elem => {
    //   return shallow(elem).html();
    // });
    // expect(result).toEqual(expected);

    //above code passes but tosses a lot of warnings
  });
  it('if there is no average rating or reviews alternate text should be displayed', () => {
    const wrapperMissingInfo = shallow(<ReviewCard taco={mockTacoNoRatingsReviews} />);
    expect(wrapperMissingInfo).toMatchSnapshot();
    console.log(wrapperMissingInfo.debug());
  });
});