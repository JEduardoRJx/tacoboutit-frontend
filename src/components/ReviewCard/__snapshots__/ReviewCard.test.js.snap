// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewCard if there is no average rating or reviews alternate text should be displayed 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(225, 225, 225, 0.3)",
      "borderRadius": 20,
      "margin": 20,
      "marginTop": 15,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.3)",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 22,
            "fontWeight": "bold",
            "justifyContent": "space-between",
            "textAlign": "center",
          }
        }
      >
        TESTING TESTING
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 22,
            "fontWeight": "bold",
            "justifyContent": "space-between",
            "textAlign": "center",
          }
        }
      >
        <Text>
          No ratings yet
        </Text>
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text>
      Be the first to add a review
    </Text>
  </View>
</View>
`;

exports[`ReviewCard should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(225, 225, 225, 0.3)",
      "borderRadius": 20,
      "margin": 20,
      "marginTop": 15,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0, 0, 0, 0.3)",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 22,
            "fontWeight": "bold",
            "justifyContent": "space-between",
            "textAlign": "center",
          }
        }
      >
        TESTING TESTING
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 22,
            "fontWeight": "bold",
            "justifyContent": "space-between",
            "textAlign": "center",
          }
        }
      >
        <Text>
          5
          /10 Avg.
        </Text>
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <View
      key="its ok1"
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "rgba(0, 0, 0, 0.3)",
          "borderBottomWidth": 2,
          "borderRadius": 30,
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        7
        /10
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        its ok
      </Text>
    </View>
    <View
      key="lame2"
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "rgba(0, 0, 0, 0.3)",
          "borderBottomWidth": 2,
          "borderRadius": 30,
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        4
        /10
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        lame
      </Text>
    </View>
    <View
      key="null3"
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "rgba(0, 0, 0, 0.3)",
          "borderBottomWidth": 2,
          "borderRadius": 30,
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        4
        /10
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      />
    </View>
    <View
      key="This taco put my son in the hospital4"
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "rgba(0, 0, 0, 0.3)",
          "borderBottomWidth": 2,
          "borderRadius": 30,
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        1
        /10
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        This taco put my son in the hospital
      </Text>
    </View>
    <View
      key="The best5"
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "rgba(0, 0, 0, 0.3)",
          "borderBottomWidth": 2,
          "borderRadius": 30,
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        10
        /10
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        The best
      </Text>
    </View>
    <View
      key="meh6"
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "rgba(0, 0, 0, 0.3)",
          "borderBottomWidth": 2,
          "borderRadius": 30,
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        4
        /10
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "textAlign": "center",
          }
        }
      >
        meh
      </Text>
    </View>
  </View>
</View>
`;
